autoload/sy/repo.vim

```diff
@@ -16,19 +16,59 @@ function! sy#repo#detect() abort
   endif
 
   for type in vcs_list
-    let [istype, diff] = sy#repo#get_diff_{type}()
-    if istype
-      return [diff, type]
-    endif
+    call sy#repo#get_diff_{type}(a:do_register)
   endfor
+    " let [istype, diff] = sy#repo#get_diff_{type}()
+    " if istype
+    "   return [diff, type]
+    " endif
+  " endfor
+
+  " return ['', 'unknown']
+endfunction
+
+" Function: s:callback_stdout_nvim {{{1
+function! s:callback_stdout_nvim(_job_id, data, _event) dict abort
+  if empty(self.stdoutbuf) || empty(self.stdoutbuf[-1])
+    let self.stdoutbuf += a:data
+  else
+    let self.stdoutbuf = self.stdoutbuf[:2]
+          \ + [self.stdoutbuf[-1] . get(a:data, 0, '')
+          \ + a:data[1:]
+  endif
+endfunction
 
-  return ['', 'unknown']
+" Function: s:callback_exit {{{1
+function! s:callback_exit(_job_id, exitcode, _event) dict abort
+  let [found_diff, diff] = a:exitcode ? [0, ''] : [1, join(self.stdoutbuf, "\n")]
+  " echomsg 'DEBUG: '. diff
+  if found_diff
+    let b:sy.type = 'git'
+  endif
+  if !self.do_register
+    let b:sy.id_top = g:id_top
+  endif
+  call sy#update_signs(diff, self.do_register)
 endfunction
 
 " Function: #get_diff_git {{{1
-function! sy#repo#get_diff_git() abort
-  let diff = s:run(g:signify_vcs_cmds.git, b:sy_info.file)
-  return v:shell_error ? [0, ''] : [1, diff]
+function! sy#repo#get_diff_git(do_register) abort
+  let cmd = s:expand_cmd(g:signify_vcs_cmds.git, b:sy_info.file)
+  let cmd = (has('win32') && &shell =~ 'cmd')  ? cmd : ['sh', '-c', cmd]
+  if exists('s:job_id_git')
+    silent! call jobstop(s:job_id_git)
+  endif
+  execute b:sy_info.chdir fnameescape(b:sy_info.dir)
+  try
+    let s:job_id_git = jobstart(cmd, {
```

Comment:
This should probably be stored in `b:sy` (or some other buffer-specific variable) to avoid overwriting things when multiple buffers are refreshed.